File: aoc21/day2/solve.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"strings"

	"strconv"
	"os"
)

func main() {
  f, err := os.Open("input.txt")
  if err != nil {
    log.Fatal(err)
  }
  defer f.Close()
  var lst []string
  scanner := bufio.NewScanner(f)
  for scanner.Scan() {
    lst = append(lst, scanner.Text())
  }
  fmt.Println(solve(lst))
}

func solve(lst []string) (int64, int64) {
  var depth1 int64 = 0
  var horiz1 int64 = 0
  var depth2 int64 = 0
  var horiz2 int64 = 0
  var aim int64 = 0
  for _, val := range lst {
    parse := strings.Split(val, " ")
    i, _ := strconv.ParseInt(parse[1], 10, 0) 
    if parse[0] == "forward" { 
      horiz1 += i 
      horiz2 += i
      depth2 += aim * i
    }
    if parse[0] == "down" { 
      depth1 += i 
      aim += i
    }
    if parse[0] == "up" { 
      depth1 -= i 
      aim -= i
    }
  }
  return horiz1 * depth1, horiz2 * depth2
}

File: 2022/4/solve.py
with open("input.txt") as f:
  lst = f.readlines()
  lst = [i.strip() for i in lst]

t = 0
t2 = 0
for i in lst:
  p1, p2 = i.split(",")
  p1 = [int(i) for i in p1.split("-")]
  p2 = [int(i) for i in p2.split("-")]
  if p1[0] <= p2[0] and p2[1] <= p1[1]:
    t += 1
  elif p2[0] <= p1[0] and p1[1] <= p2[1]:
    t+= 1
  if max(p1[0], p2[0]) <= min(p1[1], p2[1]):
    t2 += 1

print(t, t2)
